retroshare-gui/src/gui/Posted/PostedCreatePostDialog.cpp

```diff
@@ -85,37 +101,27 @@ void PostedCreatePostDialog::createPost()
 	post.mMeta.mGroupId = mGrpId;
 	post.mLink = std::string(ui->linkEdit->text().toUtf8());
 	
-	QString text;
-	text = ui->RichTextEditWidget->toHtml();
-	post.mNotes = std::string(text.toUtf8());
-
-	post.mMeta.mAuthorId = authorId;
-
-	if(!ui->titleEdit->text().isEmpty())
-	{ 
-		post.mMeta.mMsgName = std::string(ui->titleEdit->text().toUtf8());
-	}else
-	{
-		post.mMeta.mMsgName = std::string(ui->titleEditLink->text().toUtf8());
+	if(!ui->RichTextEditWidget->toPlainText().trimmed().isEmpty()) {
+		QString text;
+		text = ui->RichTextEditWidget->toHtml();
+		post.mNotes = std::string(text.toUtf8());
 	}
 
-	QByteArray ba;
-	QBuffer buffer(&ba);
+	post.mMeta.mAuthorId = authorId;
+	post.mMeta.mMsgName = std::string(ui->titleEdit->text().toUtf8());
 	
-	if(!picture.isNull())
+	if(imagebytes.size() > 0)
 	{
 		// send posted image
+		post.mImage.copy((uint8_t *) imagebytes.data(), imagebytes.size());
+	}	
 
-		buffer.open(QIODevice::WriteOnly);
-		picture.save(&buffer, "PNG"); // writes image into ba in PNG format
-		post.mImage.copy((uint8_t *) ba.data(), ba.size());
+	int msgsize = post.mLink.length() + post.mMeta.mMsgName.length() + post.mNotes.length() + imagebytes.size();

```

Comment:
When the msg is too large, it would be useful to allow the user to temporarily share the image and replace it with a retroshare link.

Reply:
(this is suggestion for later improvement. Not needed for this particular  PR).